stdlib/Test/src/Test.jl

```diff
@@ -839,6 +840,40 @@ macro test_warn(msg, expr)
     end
 end
 
+"""
+    @test_out msg expr
+
+Test whether evaluating `expr` results in [`stdout`](@ref) output that contains
+the `msg` string or matches the `msg` regular expression.  If `msg` is
+a boolean function, tests whether `msg(output)` returns `true`.  If `msg` is a
+tuple or array, checks that the screen output contains/matches each item in `msg`.
+Returns the result of evaluating `expr`.
+
+Note: For testing log statements, it is recommended to use [`@test_logs`](@ref)
+instead.
+"""
```

Comment:
Some examples might be good?

Reply:
True - I've added usage examples to `@test_output` and `@test_warn`. (Also renamed `@test_out` to `@test_output` to be more descriptive.)